homeassistant/components/overkiz/water_heater_entities/atlantic_domestic_hot_water_production_mlb_component.py

```diff
@@ -0,0 +1,182 @@
+        | WaterHeaterEntityFeature.AWAY_MODE
+        | WaterHeaterEntityFeature.ON_OFF
+    )
+    _attr_operation_list = [
+        OverkizCommandParam.PERFORMANCE,
+        OverkizCommandParam.ECO,
+        OverkizCommandParam.MANUAL,
+    ]
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self._attr_max_temp = cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MAXIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+        self._attr_min_temp = cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MINIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.MODBUSLINK_MIDDLE_WATER_TEMPERATURE
+            ),
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature corresponding to the PRESET."""
+        return cast(
+            float,
+            self.executor.select_state(OverkizState.CORE_WATER_TARGET_TEMPERATURE),
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = kwargs[ATTR_TEMPERATURE]
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_TARGET_DHW_TEMPERATURE, temperature
+        )
+
+    @property
+    def is_boost_mode_on(self) -> bool:
+        """Return true if boost mode is on."""
+        return self.executor.select_state(OverkizState.MODBUSLINK_DHW_BOOST_MODE) in (
+            OverkizCommandParam.ON,
+            OverkizCommandParam.PROG,
+        )
+
+    @property
+    def is_eco_mode_on(self) -> bool:
+        """Return true if eco mode is on."""
+        return self.executor.select_state(OverkizState.MODBUSLINK_DHW_MODE) in (
+            OverkizCommandParam.MANUAL_ECO_ACTIVE,
+            OverkizCommandParam.AUTO_MODE,
+        )
+
+    @property
+    def is_away_mode_on(self) -> bool:
+        """Return true if away mode is on."""
+        return (
+            self.executor.select_state(OverkizState.MODBUSLINK_DHW_ABSENCE_MODE)
+            == OverkizCommandParam.ON
+        )
+
+    @property
+    def current_operation(self) -> str:
+        """Return current operation."""
+        if self.is_away_mode_on:
+            return STATE_OFF
+
+        if self.is_boost_mode_on:
+            return STATE_PERFORMANCE
+
+        if self.is_eco_mode_on:
+            return STATE_ECO
+
+        if (
+            cast(str, self.executor.select_state(OverkizState.MODBUSLINK_DHW_MODE))
+            == OverkizCommandParam.MANUAL_ECO_INACTIVE
+        ):
+            return OverkizCommandParam.MANUAL
```

Comment:
This operation isn't allowed. We only allow standard operation states.

https://developers.home-assistant.io/docs/core/entity/water-heater#states

Reply:
Alas, this device does not have all the conventional operation states. It supports:
- Boost mode without setting start and end dates
- Boost mode with start and end dates set
- Away mode without setting start and end dates
- Away mode with start and end dates set
- Manual mode with ECO inactive
- Manual mode with ECO active
with #114178, all this works well for this device and, alas, I can see no way to translate these working modes in conventional operation states.

Reply:
If we can't translate a mode we'd need to make it as unknown state, ie `None`.

Reply:
How would we differentiate between all the `None` states that are unconventional?
You are trying to make the device unusable for all its users :)

Reply:
We won't be able to distinguish them but that's just how it will have to be, if you want to still represent this device in Home Assistant as a water heater.

Reply:
Okay, I will rewrite the operation state management so it only uses the conventional states in a separate pull request.
I would do it as soon as my conflicting https://github.com/home-assistant/core/pull/121801 gets merged if you allow this.

Reply:
The away mode PR doesn't seem to overlap with changing the current operation. You can open a parallel PR fixing the current operation.

Reply:
Will do. I will also fix the indentation there.
Thank you.

Reply:
done #124619